corgi/core/models.py

```diff
@@ -167,7 +167,7 @@ class Meta:
             ),
             models.UniqueConstraint(
                 name="unique_cnode_get_or_create_for_null_parent",
-                fields=("type", "purl"),
+                fields=("type", "parent", "purl"),
```

Comment:
yes I think it is true we would never want a component (parent) having 2 children componentnode(s) with the same type and purl ... I would have hoped, in terms of ingestion, we never encounter this scenario and good to put this 'guard rail' in place.